### PR TITLE
Inspector triangle count

### LINKED ISSUE #5174: Show triangle count in 3D Tiles inspector
@rahwang more enhancements for the 3D Tiles inspector, probably worth doing before https://github.com/AnalyticalGraphicsInc/cesium/issues/5156.

For this we want to see the total number of triangles that are rendered. This will operate similarly to the memory usage statistics added in https://github.com/AnalyticalGraphicsInc/cesium/pull/5050.

* For each `*3DTileContent` file add a `trianglesLength` getter, which can be modeled after `vertexMemorySizeInBytes`. It is only relevant for `Batched3DModel3DTileContent` and `Instanced3DModel3DTileContent` right now, but the others should include it.
*  Add a getter to `Model`, also modeled after `vertexMemorySizeInBytes`. The triangle count can be added when the `DrawCommand`s are created, since the `count` variable is accessible there.
* Update the statistics in `Cesium3DTileset`, like how it's done for `numberOfPointsSelected`. `numberOfTrianglesSelected` is an ok name for now.
* Add to the UI in `Cesium3DTilesInspectorViewModel`.
* Add a test for `Model`, `Batched3DModel3DTileContent`, `Instanced3DModel3DTileContent` that check the triangle length are correct. These can be similar to the `gets memory usage` tests where it loads a model or tileset and then gets the `trianglesLength` property and checks that it is correct.
* Test in `Cesium3DTileset`. You may be able to just modify all the tests that call `checkPointAndFeatureCounts` and supply the expected triangle length.

### PR DESCRIPTION
**Pull request by rahwang**
Fixes #5174 . Is this on the right track, @lilleyse ?

### CODE REVIEW

File: Source/Scene/Model.js [hunk 1]
```diff
@@ -3400,6 +3412,7 @@ define([
                     count = positions.count;
                     offset = 0;
                 }
+                model._trianglesLength = count / 3;
```
Comment (lilleyse) [inline Source/Scene/Model.js]: Do `+=` instead to handle models with multiple meshes.

**Comment by rahwang** [comment]
Ready, @lilleyse. Thanks for looking at this!

File: Source/Scene/Model.js [hunk 2]
```diff
@@ -3412,7 +3412,21 @@ define([
                     count = positions.count;
                     offset = 0;
                 }
-                model._trianglesLength = count / 3;
+
+                switch(primitive.mode) {
+                    case 0: // POINTS
```
Comment (lilleyse) [inline Source/Scene/Model.js]: Use the enums in `PrimitiveType` instead.

File: Source/Scene/Model.js [hunk 3]
```diff
@@ -3412,7 +3412,21 @@ define([
                     count = positions.count;
                     offset = 0;
                 }
-                model._trianglesLength = count / 3;
+
+                switch(primitive.mode) {
```
Comment (lilleyse) [inline Source/Scene/Model.js]: Style: `switch (`

File: Source/Scene/Model.js [hunk 4]
```diff
@@ -3412,7 +3412,21 @@ define([
...
+                    case 3: // LINE_STRIP
+                        break;
+                    case 4: // TRIANGLES
+                        model._trianglesLength += (count / 3);
+                        break;
+                    case 5: // TRIANGLE_STRIP
+                    case 6: // TRIANGLE_FAN
+                        model._trianglesLength += Math.max(count - 2, 0);
+                        break;
+                }
```
Comment (lilleyse) [inline Source/Scene/Model.js]: Now that this check is lengthier, it fits better as a separate function.

**Comment by rahwang** [comment]
Fixed!

**Comment by lilleyse** [comment]
Looks good, thanks @rahwang.

**Comment by lilleyse** [comment]
Something weird happened, when I merged github encountered an error. This PR was merged into `3d-tiles` but it wasn't closed for whatever reason.

File: Source/Scene/Model.js [hunk 5]
```diff
@@ -3342,6 +3354,22 @@ define([
         };
     }
 
+    function triangleCountFromPrimitiveIndices(primitive, indicesCount) {
+        switch (primitive.mode) {
+            case PrimitiveType.POINTS:
```
Comment (pjcozzi) [inline Source/Scene/Model.js]: All these cases that return 0 here can be removed; they are covered by the return 0 below.

**Pull request closed without merging**